vendor/github.com/swiftstack/sortedmap/llrb_tree_api.go

```diff
@@ -2,6 +2,7 @@ package sortedmap
 
 type LLRBTree interface {
 	SortedMap
+	Reset()
```

Comment:
We probably want to do this in the "common" part of the API as I think it is equally needed there.
That said, I believe I did the same thing only for the B+Tree previously - oops :-).

Reply:
Making this common seems fine.  i was being lazy!

Reply:
In retrospect, the equivalent of Reset() is actually a bit non-trivial... indeed, it is so heavyweight that the Garbage Collector would be expected to actually do a better job (i.e. I'd just say the caller should New() another one dropping the reference to the prior one)... the sync.Mutex at the top is hardly significant to it.

I rather like the distinction, now, between Reset() - meaning return the tree to the state it had just after it was created - and Discard() which actually is just a hint to cause Pass One of the Garbage Collector to know sooner that the nodes of the now unreferenced B+Tree are to be collected...